src/Orleans/Streams/Internal/StreamConsumerExtension.cs

```diff
@@ -149,15 +129,17 @@ public Task<StreamHandshakeToken> DeliverImmutable(GuidId subscriptionId, Stream
             }
             else
             {
-                // if no observer attached to the subscription, check if there's onSubscriptinChange actions defined
-                ISubscriptionChangeHandler handler;
-                if (this.onSubscriptionChangeActionMap.TryGetValue(batch.Value.GetType(), out handler))
+                // if no observer attached to the subscription, checkif the grain reference is IOnSubscriptionChangeActioner
+                if (this.onSubscriptionActioner != null)
                 {
                     //if the onAddAction attached an observer to the subscription
                     var streamProvider = this.providerRuntime.ServiceProvider
                                 .GetService<IStreamProviderManager>()
                                 .GetStreamProvider(streamId.ProviderName);
-                    await handler.InvokeOnAdd(streamId, subscriptionId, isRewindable, streamProvider);
+                   
+                    Type constructedType = typeof(OnSubscriptionActionInvoker<>).MakeGenericType(batch.Value.GetType());
```

Comment:
by " create a stream of the first concrete event type, not the base type." do you mean create a StreamSubscriptionHandle of the first concrete event type, not the base type?